lib/monad.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+// id :: a -> a
+function id(x) {
+  return x;
+}
+
+// curry :: ((a, b) -> c) -> a -> b -> c
+function curry(f, ...args1) {
+  return (...args2) => f(...args1, ...args2);
+}
+
+function withCb(func) {
```

Comment:
- We need better name and description with contact definition and examples.
- If I understand your idea, this function have following contract `withCb(func)(...args, callback)`
- So why do we need closure here, why not simply `withCb(func, ...args, callback)` ?
- And how it differs from simply calling `func(...args, callback)` ?